src/ui/map.js

```diff
@@ -2707,6 +2792,53 @@ class Map extends Camera {
         return this;
     }
 
+    /**
+     * Update the average visible elevation by sampling terrain
+     *
+     * @returns {boolean} true if elevation has changed from the last sampling
+     */
+    _updateAverageElevation(timeStamp: number, ignoreTimeout: boolean = false): boolean {
```

Comment:
nit: it might be nice to move this into a separate file to avoid adding lines to map.js

Reply:
Agreed. Since it writes private properties and executes private methods directly on the map instance, is there a logical place for it, short of just trivially moving it into another file?

Reply:
I think a new file would be fine

Reply:
Captured as follow up ticket.